pkg/storage/tsdb/caching_config.go

```diff
@@ -57,6 +58,18 @@ func (cfg *ChunksCacheConfig) Validate() error {
 	return cfg.BackendConfig.Validate()
 }
 
+func NewChunksCache(cfg ChunksCacheConfig, logger log.Logger, reg prometheus.Registerer) (chunkscache.ChunksCache, error) {
+	client, err := cache.CreateClient("chunks-cache", cfg.BackendConfig, logger, prometheus.WrapRegistererWithPrefix("cortex_", reg))
```

Comment:
We should keep consistency with the other cache metrics, we can query the same metric for all caches.

```suggestion
	client, err := cache.CreateClient("chunks-cache", cfg.BackendConfig, logger, prometheus.WrapRegistererWithPrefix("thanos_", reg))
```

Reply:
Once you will do this change, it will conflict with the chunks cache created by `CreateCachingBucket()`. My take is that they should be the same instance of chunks cache and not two different ones.

Reply:
If it's too difficult to reuse the same instance, then I would rather name this cache differently for now (but keep the `thanos_` prefix) and then renaming back to "chunks-cache" once we'll remove the other implementation.

Reply:
> then renaming back to "chunks-cache" once we'll remove the other implementation

If we plan to remove the other instance soon, then I would just follow this renaming path to keep it easy.

Reply:
I used the same instance in https://github.com/grafana/mimir/pull/4163/commits/4cef58f9770c67813339d0877afeaabd274a6abd

It's a bit repetitive because `CreateCachingBucket` now receives both the config and the  cache instance. And it's different than how the index cache is configured (only via config, without passing instance)

LMK what you think

Reply:
> If we plan to remove the other instance soon, then I would just follow this renaming path to keep it easy.

We'll need to keep it around as long as we have the non-streaming implementation.